### DISCUSSION TITLE
Common usage of `validateNestedArraySizes`

### DISCUSSION THREAD

**Discussion by Blargian** [ClickHouse/ClickHouse#62573] [Q&A] [answered]
I'm trying to figure out what the `validateNestedArraySizes` can be useful for, for documentation purposes. [2035](https://github.com/ClickHouse/clickhouse-docs/issues/2035)

For Nested data types it seems according to the documentation [here](https://clickhouse.com/docs/en/sql-reference/data-types/nested-data-structures/nested) that 
> During insertion, the system checks that they [all the component column arrays of a nested data structure] have the same length.

Query:

```sql
create table n ( a Nested( n1 int, n2 int ) ) Engine=Null;
insert into n values ( [1,2,3,4,5], [6] );
```

Response:

```response
Received exception from server (version 24.3.2):
Code: 190. DB::Exception: Received from localhost:9000. DB::Exception: Elements 'a.n1' and 'a.n2' of Nested data structure 'a' (Array columns) have different array sizes.. (SIZES_OF_ARRAYS_DONT_MATCH)

```

If it is not allowed to insert different array sizes, in what cases would it be required to first make a check using `validateNestedArraySizes`? Where is this function most likely to come in handy?

**Comment by den-crane** [comment, marked as answer]
It's just a bug that `validateNestedArraySizes` is available to SQL
